File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        //
        int low = 999;
        int res = 0;

        for (int i = 0; i < prices.size(); i++)
        {
            low = min(low, prices[i]);
            res = max(res, (prices[i] - low));
        }

        return res;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-4.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        res = 0

        start = 0
        end = len(heights) - 1
        while end > start:
            cur = min(heights[start], heights[end]) * (end-start)
            res = max(cur, res)

            if heights[start] < heights[end]:
                start += 1
            else:
                end -= 1

        return res